src/controllers/admin/mentor.controller.ts

```diff
@@ -5,65 +5,73 @@ import {
 } from '../../services/admin/mentor.service'
 import { ApplicationStatus, ProfileTypes } from '../../enums'
 import type Profile from '../../entities/profile.entity'
+import type Mentor from '../../entities/mentor.entity'
+import type { ApiResponse } from '../../types'
 
 export const mentorStatusHandler = async (
   req: Request,
   res: Response
-): Promise<void> => {
+): Promise<ApiResponse<Mentor>> => {
   try {
     const user = req.user as Profile
     const { status } = req.body
     const { mentorId } = req.params
 
     if (user.type !== ProfileTypes.ADMIN) {
-      res.status(403).json({ message: 'Only Admins are allowed' })
+      return res.status(403).json({ message: 'Only Admins are allowed' })
     } else {
       if (!(status.toUpperCase() in ApplicationStatus)) {
-        res.status(400).json({ message: 'Please provide a valid status' })
-        return
+        return res
+          .status(400)
+          .json({ message: 'Please provide a valid status' })
       }
       const { mentor, statusCode, message } = await updateMentorStatus(
         mentorId,
         status
       )
-      res.status(statusCode).json({ mentor, message })
+      return res.status(statusCode).json({ mentor, message })
     }
   } catch (err) {
     if (err instanceof Error) {
       console.error('Error executing query', err)
-      res
+      return res
         .status(500)
         .json({ error: 'Internal server error', message: err.message })
     }
+
+    throw err
   }
 }
 
 export const getAllMentorsByStatus = async (
   req: Request,
   res: Response
-): Promise<void> => {
+): Promise<ApiResponse<Mentor>> => {
   try {
     const user = req.user as Profile
     const status: ApplicationStatus | undefined = req.query.status as
       | ApplicationStatus
       | undefined
 
     if (user.type !== ProfileTypes.ADMIN) {
```

Comment:
Here as well,


```
    if (user.type !== ProfileTypes.ADMIN) {
      return res.status(403).json({ message: 'Only Admins are allowed' })
    }

    if (status && !(status?.toUpperCase() in ApplicationStatus)) {
      return res.status(400).json({ message: 'Please provide a valid status' })
    }

    const { mentors, statusCode, message } = await getAllMentors(status)
    return res.status(statusCode).json({ mentors, message })
```

Reply:
@Shrenik0321 Can you check other conditions as well? we don't have to nest them with the returns.